runners/direct-java/src/main/java/org/apache/beam/runners/direct/WatermarkManager.java

```diff
@@ -1537,6 +1680,21 @@ public boolean equals(Object other) {
           && Objects.equals(this.setTimers, that.setTimers)
           && Objects.equals(this.deletedTimers, that.deletedTimers);
     }
+
+    @Override
+    public String toString() {
+      return "TimerUpdate(key="
```

Comment:
`MoreObject.toStringHelper` is a great utility for this sort of thing.

Reply:
:+1: